cmd/frontend/graphqlbackend/search_results_test.go

```diff
@@ -106,21 +100,17 @@ func TestSearchResults(t *testing.T) {
 		mockDecodedViewerFinalSettings = &schema.Settings{}
 		defer func() { mockDecodedViewerFinalSettings = nil }()
 
-		var calledReposListMinimalRepos bool
-		database.Mocks.Repos.ListMinimalRepos = func(_ context.Context, op database.ReposListOptions) ([]types.MinimalRepo, error) {
-			calledReposListMinimalRepos = true
-
-			// Validate that the following options are invariant
-			// when calling the DB through Repos.ListMinimalRepos, no matter how
-			// many times it is called for a single Search(...) operation.
-			assertEqual(t, op.LimitOffset, limitOffset)
-			assertEqual(t, op.IncludePatterns, []string{"r", "p"})
+		repos := dbmock.NewMockRepoStore()
+		repos.ListMinimalReposFunc.SetDefaultHook(func(ctx context.Context, opt database.ReposListOptions) ([]types.MinimalRepo, error) {
+			// Validate that the following options are invariant when calling the DB through
+			// Repos.ListMinimalRepos, no matter how many times it is called for a single
+			// Search(...) operation.
+			assert.Equal(t, wantLimitOffset, opt.LimitOffset)
```

Comment:
nit: the removed `assertEqual` function is more equivalent to `require.Equal` since it calls `t.Fatalf()` rather than `t.Errorf()`

```suggestion
			require.Equal(t, wantLimitOffset, opt.LimitOffset)
```